### ISSUE TITLE
Rest API products - orderby price

### ISSUE THREAD

**Issue by steffanhalv** [woocommerce/woocommerce#16215]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce
- [x] This is not a usage question (Those should be directed to the [community](https://wordpress.org/support/plugin/woocommerce), unless this is a question about a premium plugin in which you should [use the helpdesk](https://woocommerce.com/my-account/tickets/) for official extensions or contact the author of 3rd party extensions)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

`orderby=price` in query params for products list

-> http://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products

(It is not supported in documentation, which I find strange )

Also tried to manually add the previous filter version:

```
function wc_rest_api_filter_support() {
	foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
		add_filter( 'woocommerce_rest_' . $post_type->name . '_object_query', 'wc_rest_api_filter_support_addon', 10, 2 );
	}
}

/**
 * @param array $args The query arguments
 * @param WP_REST_Request $request Full details about the request
 * @return array $args
 **/
function wc_rest_api_filter_support_addon( $args, $request ) {
	if ( empty( $request['filter'] ) || ! is_array( $request['filter'] ) ) {
		return $args;
	}
	$filter = $request['filter'];
	global $wp;
	$vars = apply_filters( 'woocommerce_rest_query_vars', $wp->public_query_vars );
	foreach ( $vars as $var ) {
		if ( isset( $filter[ $var ] ) ) {
			$args[ $var ] = $filter[ $var ];
		}
	}
	return $args;
}
```

And add filter[orderby]=meta_value_num&filter[orderby_meta_key]=_regular_price

or

filter[orderby]=meta_value_num&filter[orderby_meta_key]=regular_price

or

filter[orderby]=meta_value_num&filter[orderby_meta_key]=price

etc. But its never ordered by price.

## Expected behavior and actual behavior

Expected products to be ordered by price, how to do this?

This should have been a primary orderby option.

All of these: regular_price, sale_price, total_sales, average_rating, rating_count, sku, downloads, weight, dimensions and menu_order has several use cases to be included as an orderby option. At least regular_price and sale_price

## Environment

<details>
```
Grab the system status report from WooCommerce > System Status and paste it here.
```
</details>

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently

**Comment by mikejolley** [comment]
This is not supported in the v2 API. Supported values are:

`
Sort collection by object attribute. Options: date, id, include, title and slug. Default is date.
`

Filter is also not supported by WordPress either.

Price will probably be added in v3 as we expand out CRUD and search helpers. Those are not ready for products yet. https://github.com/woocommerce/woocommerce/issues/13645

**Comment by mattwomple** [comment]
Is this possible now that #13645 is resolved with pull #16291 ? I gather that the `orderby` param of [List all products](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products) doesn't directly support any other options than `date, id, include, title, slug`, but I'm not sure if this is supposed to be supported behind the `search` or `attribute` parameters.

Somewhat related instructions that I've tried to repurpose (unsuccessfully) include:

- [Custom sorting options (asc/desc)](https://docs.woocommerce.com/document/custom-sorting-options-ascdesc/)
- [Adding Custom Parameter Support](https://github.com/woocommerce/woocommerce/wiki/wc_get_products-and-WC_Product_Query#adding-custom-parameter-support)

If this is possible (I'm interested in ordering by product `weight`), is there an example from which I could draw?
